src/lib/swapper/swappers/ThorchainSwapper/utils/getLimit/getLimit.ts

```diff
@@ -75,8 +71,14 @@ export const getLimit = async ({
     )
   }
 
-  const sellFeeAssetUsdRate = await getUsdRate(deps.daemonUrl, sellAssetChainFeeAssetId)
-  const buyAssetUsdRate = await getUsdRate(deps.daemonUrl, buyAssetId)
+  const maybeSellFeeAssetUsdRate = await getUsdRate(deps.daemonUrl, sellAssetChainFeeAssetId)
+  if (maybeSellFeeAssetUsdRate.isErr()) return Err(maybeSellFeeAssetUsdRate.unwrapErr())
+  const sellFeeAssetUsdRate = maybeSellFeeAssetUsdRate.unwrap()
+
+  const maybeBuyAssetUsdRate = await getUsdRate(deps.daemonUrl, buyAssetId)
+  if (maybeBuyAssetUsdRate.isErr()) return Err(maybeBuyAssetUsdRate.unwrapErr())
+  const buyAssetUsdRate = maybeBuyAssetUsdRate.unwrap()
```

Comment:
This is ugly, though I'm not sure how else we'd do it without having deeply nested path hell.

Reply:
Yeah, until I find working heuristics to deal with nested async values, that's our best bet unfortunately